ddtrace/internal/ci_visibility/git_client.py

```diff
@@ -166,6 +178,11 @@ def _run_protocol(
         log.setLevel(log_level)
         _metadata_upload_status.value = METADATA_UPLOAD_STATUS.IN_PROCESS
         try:
+            if not cls._get_git_dir(cwd=cwd):
+                log.debug("Missing .git directory; skipping git metadata upload")
+                _metadata_upload_status.value = METADATA_UPLOAD_STATUS.FAILED
+                return
```

Comment:
I think it may make more sense to do this during `__init__()` (and have the upload return early if there's no directory, for example) , or during `upload_git_metadata()` prior to us spinning up the worker process (since it's not going to do anything anyway).

Reply:
I recall considering this but going the other route for consistency with the way that the subprocess owns the setting the status value, or perhaps it made the unit test a bit easier to write; I'm not sure. I'll take another pass.